x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_rule_execution_events.tsx

```diff
@@ -6,25 +6,34 @@
  */
 
 import { useQuery } from 'react-query';
+import { GetAggregateRuleExecutionEventsResponse } from '../../../../../common/detection_engine/schemas/response';
 import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
 import { fetchRuleExecutionEvents } from './api';
 import * as i18n from './translations';
 
-// TODO: https://github.com/elastic/kibana/pull/121644 clean up
-export const useRuleExecutionEvents = (ruleId: string) => {
+interface UseRuleExecutionEventsArgs {
+  ruleId: string;
+  start: string;
+  end: string;
+  filters?: string;
+}
+
+export const useRuleExecutionEvents = ({
+  ruleId,
+  start,
+  end,
+  filters,
+}: UseRuleExecutionEventsArgs) => {
   const { addError } = useAppToasts();
 
-  return useQuery(
-    'ruleExecutionEvents',
+  return useQuery<GetAggregateRuleExecutionEventsResponse>(
+    ['ruleExecutionEvents', start, end, filters],
```

Comment:
`ruleId` should also be in the query key. Otherwise, events from one rule could appear on the other's rule page as their cache keys match by default. We should include all request params in the query key as a rule of thumb.

Reply:
Whoops! Good catch, thank you! 🙂

Reply:
My bad, it was missing in the hook that I added in the previous PR. Sorry Garrett, thanks Dmitrii 🙂

Reply:
Fixed in https://github.com/elastic/kibana/pull/124194